plugins/nf-cid-h2/src/main/nextflow/data/cid/h2/H2CidStore.groovy

```diff
@@ -121,11 +133,34 @@ class H2CidStore implements CidStore {
         return new H2CidHistoryLog(dataSource)
     }
 
+    @Override
+    List<CidSerializable> search(String queryString) {
+        final results= new LinkedList<CidSerializable>()
+        try(final sql=new Sql(dataSource)) {
+            sql.eachRow("SELECT metadata FROM cid_file WHERE JSON_MATCH(metadata, ?)", List.<Object>of(queryString)) { row ->
+                results.add(encoder.decode(toValue(row['metadata']) as String))
+            }
+        }
+        return results
+    }
+    /**
+     * JSON_MATCH implementation for h2
+     * @param jsonString
+     * @param queryString
+     * @return
+     */
+    static boolean matchesJsonQuery(String jsonString, String queryString) {
+        def json = new JsonSlurper().parseText(jsonString)
```

Comment:
Let's prefer the new Json encoder whenever possible

Reply:
Fixed